lib/rules/lightness-notation/index.mjs

```diff
@@ -0,0 +1,183 @@
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, {
+			actual: primary,
+			possible: ['percentage', 'number'],
+		});
+
+		if (!validOptions) return;
+
+		root.walkDecls((decl) => {
+			if (!HAS_LIGHTNESS_COLOR_FUNC.test(decl.value)) return;
+
+			let needsFix = false;
+			const parsedValue = valueParser(getDeclarationValue(decl));
+
+			parsedValue.walk((node) => {
+				if (node.type !== 'function') return;
+
+				if (!LIGHTNESS_FUNCS.has(node.value.toLowerCase())) return;
+
+				const lightness = findLightness(node);
+
+				if (!lightness) return;
+
+				const { value } = lightness;
+
+				if (!isStandardSyntaxValue(value)) return;
+
+				if (!isPercentage(value) && !isNumber(value)) return;
+
+				if (primary === 'percentage' && isPercentage(value)) return;
+
+				if (primary === 'number' && isNumber(value)) return;
+
+				const functionName = node.value;
+				const fixed =
+					primary === 'percentage'
+						? asPercentage(value, functionName)
+						: asNumber(value, functionName);
+				const unfixed = value;
+
+				if (context.fix) {
+					lightness.value = fixed;
+					needsFix = true;
+
+					return;
+				}
+
+				const valueIndex = declarationValueIndex(decl);
+
+				report({
+					message: messages.expected,
+					messageArgs: [unfixed, fixed],
+					node: decl,
+					index: valueIndex + lightness.sourceIndex,
+					endIndex: valueIndex + lightness.sourceEndIndex,
+					result,
+					ruleName,
+				});
+			});
+
+			if (needsFix) {
+				setDeclarationValue(decl, parsedValue.toString());
+			}
+		});
+	};
+};
+
+/**
+ * @param {string} value
+ * @param {import('postcss-value-parser').FunctionNode['value']} func
+ */
+function asPercentage(value, func) {
+	let num = parseFloat(value);
+
+	if (LIGHTNESS_ZERO_TO_HUNDRED_RANGE_FUNCS.has(func)) {
+		return `${num}%`;
+	}
+
+	if (LIGHTNESS_ZERO_TO_ONE_RANGE_FUNCS.has(func)) {
+		num *= 100;
+	}
+
+	if (Number.isInteger(num)) {
+		return `${num}%`;
+	}
+
+	return `${roundToNumberOfDigits(num, value)}%`;
+}
+
+/**
+ * @param {string} value
+ * @param {import('postcss-value-parser').FunctionNode['value']} func
+ */
+function asNumber(value, func) {
+	const dimension = valueParser.unit(value);
+
+	if (!dimension) {
+		throw new TypeError(`The "${value}" value must have a unit`);
+	}
+
+	if (LIGHTNESS_ZERO_TO_ONE_RANGE_FUNCS.has(func)) {
+		const num = Number(dimension.number) / 100;
+
+		return `${roundToNumberOfDigits(num, value)}`;
+	}
+
+	return dimension.number;
+}
+
+/**
+ * @param {number} num
+ * @param {string} value
+ */
+function roundToNumberOfDigits(num, value) {
+	return num.toPrecision(value.length - 2);
+}
+
+/**
+ * @param {import('postcss-value-parser').FunctionNode} node
+ */
+function findLightness(node) {
+	const args = node.nodes.filter(({ type }) => type === 'word' || type === 'function');
+	const value = node.value.toLowerCase();
+
+	if (LIGHTNESS_FUNCS.has(value)) {
+		return args[0];
+	}
+
+	return undefined;
```

Comment:
[suggest] How about simplifying `findLightness()` like this? `LIGHTNESS_FUNCS.has()` check seems duplicate at [L50](https://github.com/stylelint/stylelint/pull/7366/files#diff-e55c9791f54478934c2ad8ddd81e2913441c88b5ba16638f72d10ba02214c11aR50).

```suggestion
function findLightness({ nodes }) {
	return nodes.find(({ type }) => type === 'word' || type === 'function');
```